indolaw-parser/parser_utils.py

```diff
@@ -847,29 +847,48 @@ def clean_squashed_page_numbers(law: List[str]) -> List[str]:
     print_section_header('CLEANING SQUASHED PAGE NUMBER...')
 
     new_law = []
-    for idx, line in enumerate(law):
+    
+    idx = 0
+    last_idx = len(law)
+    last_input_idx = 0
+
+    while idx < last_idx:
+        line = law[idx]
+
+    #for idx, line in enumerate(law):
         result = re.split(PAGE_NUMBER_REGEX, line)
         if len(result) == 1:
             new_law.append(line)
         elif len(result) > 1:
+            
             print_line()
             print(f'{idx} / {len(law)}')
             print(line)
             print_line()
 
             pyperclip.copy(line)
             print('Does this line have a page number squashed onto the end?')
-            print_yes_no()
+            print_yes_no_undo()
             user_input = input()
 
             # user_input = 'y'
             if user_input == 'y':
                 new_law.append(''.join(result[:-2]))
             elif user_input == 'n':
                 new_law.append(line)
+            elif user_input == 'z':
+                for delete_idx in range(idx - 1, last_input_idx - 1, -1):
+                    new_law.pop(delete_idx)
```

Comment:
qq: instead of passing an argument to `new_law.pop` can't we just call `pop()` w/o arguments since we basically want to remove all lines from most recent until `last_input_idx`

![](https://media2.giphy.com/media/WxIXZWVuZFRmlgwu8b/giphy.gif)

Reply:
😲 that's true! Okay will do!